src/checkbox_bubble.js

```diff
@@ -0,0 +1,227 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from "blockly/core";
+
+/**
+ * A checkbox shown next to reporter blocks in the flyout.
+ * @implements {Blockly.IBubble}
+ * @implements {Blockly.IRenderedElement}
+ */
+export class CheckboxBubble {
+  /**
+   * Size of a checkbox next to a variable reporter.
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_SIZE = 25;
+
+  /**
+   * Amount of touchable padding around reporter checkboxes.
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_TOUCH_PADDING = 12;
+
+  /**
+   * SVG path data for checkmark in checkbox.
+   * @type {string}
+   * @const
+   */
+  static CHECKMARK_PATH =
+    "M" +
+    CheckboxBubble.CHECKBOX_SIZE / 4 +
+    " " +
+    CheckboxBubble.CHECKBOX_SIZE / 2 +
+    "L" +
+    (5 * CheckboxBubble.CHECKBOX_SIZE) / 12 +
+    " " +
+    (2 * CheckboxBubble.CHECKBOX_SIZE) / 3 +
+    "L" +
+    (3 * CheckboxBubble.CHECKBOX_SIZE) / 4 +
+    " " +
+    CheckboxBubble.CHECKBOX_SIZE / 3;
+
+  /**
+   * Size of the checkbox corner radius
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_CORNER_RADIUS = 5;
+
+  /**
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_MARGIN = 12;
+
+  /**
+   * Total additional width of a row that contains a checkbox.
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_SPACE_X =
+    CheckboxBubble.CHECKBOX_SIZE + 2 * CheckboxBubble.CHECKBOX_MARGIN;
+
+  svgRoot;
```

Comment:
Add doc--here and for all methods that are missing it.

Reply:
Done